src/doc/rustc/src/check-cfg.md

```diff
@@ -1,34 +1,51 @@
 # Checking conditional configurations
 
-`rustc` accepts the `--check-cfg` option, which specifies whether to check conditions and how to
-check them. The `--check-cfg` option takes a value, called the _check cfg specification_.
-This specification has one form:
+`rustc` supports checking that every _reachable_[^reachable] `#[cfg]` matches a list of the
+expected config names and values.
 
-1. `--check-cfg cfg(...)` mark a configuration and it's expected values as expected.
+This can help with verifying that the crate is correctly handling conditional compilation for
+different target platforms or features. It ensures that the cfg settings are consistent between
+what is intended and what is used, helping to catch potential bugs or errors early in the
+development process.
 
-*No implicit expectation is added when using `--cfg`. Users are expected to
-pass all expected names and values using the _check cfg specification_.*
+In order to accomplish that goal, `rustc` accepts the `--check-cfg` flag, which specifies
+whether to check conditions and how to check them.
 
-## The `cfg(...)` form
+> **Note:** No implicit expectation is added when using `--cfg`. Users are expected to
+pass all expected names and values using the _check cfg specification_.
 
-The `cfg(...)` form enables checking the values within list-valued conditions. It has this
-basic form:
+[^reachable]: `rustc` promises to at least check reachable `#[cfg]`, and while non-reachable
+`#[cfg]` are not currently checked, they may well be checked in the future without it being a
+breaking change.
+
+## Specifying and configuring expected names and values
```

Comment:
I'm not sure I understand the difference between "specifying" and "configuring". They seem to be essentially the same, so I'm not sure why there is a distinction here.

```suggestion
## Specifying expected names and values
```

Reply:
The difference is only for `cfg(any())` and `cfg(name, values(any()))` which doesn't specify a list of expected names or values but expect **every** names or values. Which is more a of configuration (`cfg(any())` flips the default from nothing is expected to everything is expected, same for the other one).

But, I agree, the distinction doesn't seems very useful. Removed.